### DISCUSSION TITLE
about useProxy and useSnapshot

### DISCUSSION THREAD

**Discussion by zhangfanhang** [pmndrs/valtio#883] [Q&A] [answered]
Great state library!!! I'd like to ask why there is no description and explanation of useProxy in the official documentation. What's the difference between this usage and usesnapshot seen in many code bases?

**Comment by dai-shi** [comment, marked as answer]
README is the official documentation and describes useProxy. Also #869 is work-in-progress by @ScreamZ.

useProxy is based on useSnapshot. If you read the code you will see it's very small.
I hesitated to recommend useProxy before introducing useSnapshot, because the mental model is still important. Thus, `useSnapshot` is described first and then `useProxy`.
